pkg/apis/apps/v1/defaults_test.go

```diff
@@ -304,7 +308,8 @@ func TestSetDefaultStatefulSet(t *testing.T) {
 					UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
 						Type: appsv1.RollingUpdateStatefulSetStrategyType,
 						RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
-							Partition: &notTheDefaultPartition,
+							Partition:      &notTheDefaultPartition,
+							MaxUnavailable: &notTheDefaultMaxUnavailable,
```

Comment:
since it is disabled in this test case, I would expect this to end up nil? If some other part of the system clears it then maybe don't test this here, as it is confusing?

Reply:
agree, removing